azure-pipelines.yml

```diff
@@ -57,42 +50,58 @@ jobs:
   - pwsh: build/Update-Headers.ps1 -Verify
     displayName: Verify headers
 
+  # Cache NuGet Packages across pipeline runs
```

Comment:
Caching NuGet packages is absolutely overkill, plus I really don't want to add thousands of unmaintainable .lock files that could very easily cause all sorts of issues. It's perfectly fine to just restore NuGet packages normally whenever a build is run. Can you please rebase and drop the last commit? Thank you 🙂

Reply:
I thought so too. I didn't like it either. I'll drop it.

> Caching NuGet packages is absolutely overkill

**NO** It's not overkill. With caching, I could see the build time drop significantly in subsequent builds. It's definitely worth it. But not when it comes to the maintainability of the project. So, to not have multiple lock files, there's a central lock file (_single lock file for entire repo_) feature that is coming to NuGet CPM. We can add caching to the CPM PR when it's available in time.

Reply:
Yes, caching reduces the build time, but even without it, we have no issues. The whole CI still just takes a few minutes, so it's just not worth the extra complexity. It's not like we have hundreds of CIs that take hours to run, like the runtime. Even without any kind of caching, a full CI run just takes like 10 minutes, so I really don't see the value here for all that extra complexity. I'm fine with caching the tools since that's self-contained in the YAML script, but I'm not ok with adding .lock files that would need to be kept in sync, it's just not worth it in our case 🙂